lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -2128,6 +2128,23 @@ PullbackCloner::PullbackCloner(VJPCloner &vjpCloner)
 
 PullbackCloner::~PullbackCloner() { delete &impl; }
 
+static SILValue getArrayValue(ApplyInst *ai) {
+  SILValue arrayValue;
+  for (auto use : ai->getUses()) {
+    auto *dti = dyn_cast<DestructureTupleInst>(use->getUser());
+    if (!dti)
+      continue;
+    assert(!arrayValue && "Array value already found");
+    // The first `destructure_tuple` result is the `Array` value.
+    arrayValue = dti->getResult(0);
+#ifdef NDEBUG
```

Comment:
Consider using `ASSERT` / `DEBUG_ASSERT` from swift/Basic/Assertions.h instead of conditional asserts, etc. Overall, the intention is to get rid of conditional asserts in new code (and avoid `#ifdef NDEBUG`).

Reply:
I've put this `break;` under `DEBUG_ASSERT_enabled` condition since having the check present increases iterations count and this should probably not be done in non-debug builds. Please let me know if this looks reasonable for you.